src/goose/brokers/redis/batch.clj

```diff
@@ -0,0 +1,27 @@
+(ns goose.brokers.redis.batch
+  (:require
+    [goose.brokers.redis.commands :as redis-cmds]
+    [goose.brokers.redis.retry :as retry]
+    [goose.defaults :as d]))
+
+(defn metadata-update [next]
```

Comment:
lets rename this. `update-stats` maybe?

Reply:
Did we decide on calling `metadata` as `stats`?
I think `stats` is similarly vague as `metadata`.

Reply:
`stats` is definitely not vague. Clearly shows that this data represents batch statistics.
But as we want to add callback-fn, id etc as well, batch state also works

Reply:
Noted. I have renamed the function to `update-state`